enginetest/enginetests.go

```diff
@@ -1594,6 +1594,18 @@ func TestStoredProcedures(t *testing.T, harness Harness) {
 		ctx := NewContext(harness)
 		ctx.SetCurrentDatabase("")
 
+		for _, script := range queries.NoDbProcedureTests {
+			if script.Expected != nil || script.SkipResultsCheck {
+				expectedResult := script.Expected
+				if script.SkipResultsCheck {
+					expectedResult = nil
+				}
+				TestQueryWithContext(t, ctx, e, harness, script.Query, expectedResult, nil, nil)
+			} else if script.ExpectedErr != nil {
+				AssertErrWithCtx(t, e, harness, ctx, script.Query, script.ExpectedErr)
+			}
+		}
+
```

Comment:
I left a comment later about trying out `queries.NoDbProcedureTests` as `[]ScriptTest` instead of just assertions, thinking that it would simplify this code so it would look just like the blocks above that call `TestScript` on different ProcedureShowFoo tests, but now I'm realizing that those probably select a database by default. 

Not a big deal, but if you did want to tidy this up so it looks more standard like the blocks above... I think you could use a hack like this to get no db selected (I found [this idea on StackOverflow](https://stackoverflow.com/questions/2510276/how-do-i-undo-git-reset#:~:text=So%2C%20to%20undo%20the%20reset,to%20search%20through%20the%20reflog%20.)):
```
CREATE DATABASE temporaryDb;
USE temporaryDb;
DROP DATABASE temporaryDb;
```

Not a huge deal, but might clean this test function up just a little bit. If you really wanted to be fancy, you could potentially move these existing NoDb tests into that same NoDbSelected []ScriptTest and then have this test function be super standardized.

Reply:
That was my first attempt, to use the "CREATE/USE/DROP" approach in the SetUpScript to "clear out" the database. (And this approach to unsetting the DB works while using dolt manually.)
But when running through our test harness, the "CREATE/USE/DROP" approach does not work, `select database()` still returns `mydb`. Seems like it's an issue with the our test code, but haven't dug in. 

Manually creating a context and calling `ctx.SetCurrentDatabase("")` seems to work to unset the DB, however. I can try to update `TestScriptWithEngine` to optionally set the DB to empty, but that seems like a farther-reaching change.

Thoughts?

Reply:
Discussed offline: keeping the tests as they are, but if we end up needing more no-DB tests like these, we should rework the harness a bit and make it easier to test these cases.